### PR TITLE
Uses GSRN as entityId for activity log entries in certificates

### PR DESCRIPTION
**Pull request by tnickelsen**

### CODE REVIEW

File: domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs [hunk 1]
```diff
@@ -157,22 +157,24 @@ public class ContractsController : ControllerBase
...
 
+        var contract = await service.GetById(id, meteringPointOwner, cancellationToken);
+
         return result switch
         {
             NonExistingContract => NotFound(),
             MeteringPointOwnerNoMatch => Forbid(),
             EndDateBeforeStartDate => ValidationProblem("EndDate must be after StartDate"),
-            SetEndDateResult.Success => await LogChangedEndDateAndReturnOk(activityLogEntryRepository, user, id),
+            SetEndDateResult.Success => await LogChangedEndDateAndReturnOk(activityLogEntryRepository, user, contract!.GSRN),
```
Comment (sahma19) [inline domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs:167]: You are going through the contract to get the gsrn? I thought you wanted to avoid that?

Also why is the null warning suppressed? Can't you make it null safe?

File: domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs [hunk 2]
```diff
@@ -157,22 +157,24 @@ public class ContractsController : ControllerBase
             newEndDate,
             cancellationToken);
 
+        var contract = await service.GetById(id, meteringPointOwner, cancellationToken);
```
Comment (sahma19) [inline domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs:160]: Why is this needed?

File: domains/certificates/Shared/DataContext/Migrations/20240216110232_ActivityLogEntityIdIsNowAString.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+using System;
```
Comment (sahma19) [inline domains/certificates/Shared/DataContext/Migrations/20240216110232_ActivityLogEntityIdIsNowAString.cs:1]: Do we have a migration test for this?

File: domains/certificates/Shared/DataContext/Migrations/ApplicationDbContextModelSnapshot.cs [hunk 4]
```diff
@@ -193,8 +193,9 @@ protected override void BuildModel(ModelBuilder modelBuilder)
                     b.Property<int>("ActorType")
                         .HasColumnType("integer");
 
-                    b.Property<Guid>("EntityId")
-                        .HasColumnType("uuid");
+                    b.Property<string>("EntityId")
+                        .IsRequired()
```
Comment (sahma19) [inline domains/certificates/Shared/DataContext/Migrations/ApplicationDbContextModelSnapshot.cs:197]: Why is it marked as required? It wasn't before?

File: domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs [hunk 5]
```diff
@@ -62,7 +62,7 @@ private async Task DeleteOldTransferAgreementProposals(DateTimeOffset olderThan,
 
         var activityLogEntries = oldProposals.Select(proposal => ActivityLogEntry.Create(Guid.Empty, ActivityLogEntry.ActorTypeEnum.System,
             string.Empty, string.Empty, string.Empty, ActivityLogEntry.EntityTypeEnum.TransferAgreementProposal,
-            ActivityLogEntry.ActionTypeEnum.Expired, proposal.Id));
+            ActivityLogEntry.ActionTypeEnum.Expired, proposal.Id.ToString()));
```
Comment (sahma19) [inline domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs:65]: I thought it was possible to cast the guid to a string, automatically?

**Review by sahma19 — CHANGES_REQUESTED** [review]

File: domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs (hunk 1 above)
Comment (tnickelsen) [reply domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs:167]: Why would I avoid that? It's the only way to get the GSRN

I'm not suppressing any warnings, just telling the compiler I'm sure it's not null here since I jsut updated it

File: domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs (hunk 2 above)
Comment (tnickelsen) [reply domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs:160]: To get the GSRN number

File: domains/certificates/Shared/DataContext/Migrations/ApplicationDbContextModelSnapshot.cs (hunk 4 above)
Comment (tnickelsen) [reply domains/certificates/Shared/DataContext/Migrations/ApplicationDbContextModelSnapshot.cs:197]: Because it's not a nullable field

File: domains/certificates/Shared/DataContext/Migrations/20240216110232_ActivityLogEntityIdIsNowAString.cs (hunk 3 above)
Comment (tnickelsen) [reply domains/certificates/Shared/DataContext/Migrations/20240216110232_ActivityLogEntityIdIsNowAString.cs:1]: Yes

File: domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs (hunk 5 above)
Comment (tnickelsen) [reply domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs:65]: No no, you have to ToString() it

File: domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs (hunk 1 above)
Comment (sahma19) [reply domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs:167]: ok

File: domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs (hunk 2 above)
Comment (sahma19) [reply domains/certificates/Query.API/API/Query.API/v2023_01_01/Controllers/ContractsController.cs:160]: ok

File: domains/certificates/Shared/DataContext/Migrations/ApplicationDbContextModelSnapshot.cs (hunk 4 above)
Comment (sahma19) [reply domains/certificates/Shared/DataContext/Migrations/ApplicationDbContextModelSnapshot.cs:197]: ok

File: domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs (hunk 5 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs:65]: ok

**Review by sahma19 — DISMISSED** [review]

**Review by sahma19 — APPROVED** [review]
